provider/kv_test.go

```diff
@@ -410,6 +410,14 @@ func TestKVLoadConfig(t *testing.T) {
 					Key:   "traefik/backends/backend.with.dot.too/servers/server.with.dot/weight",
 					Value: []byte("0"),
 				},
+				{
+					Key:   "traefik/backends/backend.with.dot.too/servers/server.with.dot.without.url",
+					Value: []byte(""),
+				},
+				{
+					Key:   "traefik/backends/backend.with.dot.too/servers/server.with.dot.without.url/weight",
+					Value: []byte("0"),
+				},
```

Comment:
I'd change the `Mock` struct to separate the currently single error into one for `Get` and one for `List` calls. We'll then have to adjust all tests that test error cases, but I think it's worth it: We have to improve our test coverage in the future anyway, and more fine-grained error distinctions are most likely going to be needed.